featuretools/entityset/deserialize.py

```diff
@@ -1,11 +1,18 @@
 import json
 import os
+import tarfile
+from pathlib import Path
 
+import boto3
 import pandas as pd
+import s3fs
+from backports import tempfile
```

Comment:
Since TemporaryDirectory was added in Python 3.4 and we do not support python 3 versions older than 3.5,  the backport should only be necessary for Python 2.7.  We can use the `is_python_2` utils function to decide which way to import tempfile and only require the backports.tempfile module for Python 2 installs